lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -676,6 +816,28 @@ def __init__(self, fullname, path_list):
         if not self._redirect:
             raise ImportError('not redirected, go ask path_hook')
 
+    def get_filename(self, fullname):
+        return None
```

Comment:
Should this be something like `<synthetic package/module>`? I know that `eval()`/`python -c` does something like this.

Reply:
Hm... `'<ansible_synthetic_redirection>'`? Feel free to bikeshed

Reply:
I don't have any preference. The only thing is that it probably shouldn't use underscores instead of plain old whitespace. It's not an identifier, after all. But I won't insist on changing this.

Reply:
Okay, I'll fix that.